ui/src/main/java/edu/wpi/grip/ui/util/NodeControllerObservableListMap.java

```diff
@@ -0,0 +1,196 @@
+package edu.wpi.grip.ui.util;
+
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import edu.wpi.grip.ui.Controller;
+import javafx.application.Platform;
+import javafx.collections.FXCollections;
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
+import javafx.scene.Node;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public final class NodeControllerObservableListMap<C extends Controller, N extends Node> implements Map<C, N> {
```

Comment:
This REALLY needs to be documented fully.
This is an awesome utility class that I created and it does a ton of heavy lifting for you. I think the sort function could be used in the preview pane to make the entire class much simpler.